pytorch_lightning/metrics/utils.py

```diff
@@ -264,3 +264,21 @@ def class_reduce(
     raise ValueError(
         f"Reduction parameter {class_reduction} unknown." f" Choose between one of these: {valid_reduction}"
     )
+
+
+def _stable_1d_sort(x: torch):
+    """
+    Stable sort of 1d tensors. Pytorch defaults to a stable sorting algorithm
+    if number of elements are larger than 2048. This function pads the tensors,
+    makes the sort and returns the sorted array (with the padding removed)
+    See this discussion: https://discuss.pytorch.org/t/is-torch-sort-stable/20714
+    """
+    N = 2049
```

Comment:
can we move this as a parameter with default?

Reply:
Done :]